templates/webpack.config.js

```diff
@@ -1,17 +1,46 @@
-const configFunc = require("@thc/webpack-react/lib/webpack-config");
-const { prepareEntries } = require("@thc/webpack-react/lib/utils");
-//const path = require('path');
+const {
+    handleAssets,
+    handleJs,
+    handleCss,
+    emptyConf,
+    configEntries,
+    addHotReload,
+    addHtmlIndex,
+    miscOptions,
+    optimize,
+    configOutput,
+    generateSourcemap,
+    utilities
+} = require("@thc/webpack-react/lib/blocks");
+
+const { envDefaults } = require("@thc/webpack-react/lib/utils");
 
 module.exports = (processEnv, argv) => {
-    const env = processEnv || process.env;
-    const config = configFunc(env, argv);
+    const env = envDefaults(processEnv);
+    // Every function accepts a config object
+    // Object is gonna be shallow merged with default conf
+    const configFunctions = [
```

Comment:
If you look at my comment on `emptyConf`, do something like:

```js
const config = makeEmptyConfig();

const configurator = createConfigurator(env, argv, [
    handleJs(),
    handleCss(),
    handleAssets(),
    configEntries(), //Default entry object is : { main : "./src/app.js" }
    addHotReload(),
    addHtmlIndex(),
    miscOptions(),
    optimize(),
    configOutput(),
    generateSourcemap(),
    utilities() // Must be after output => depends on config.output.path
]);

return configurator(config); 
```

and expose from the lib:

```js
module.exports = function createConfigurator(env, argv, configFunctions) {
    return compose(configFunctions.filter(Boolean).map(confFunc => conFunc(env, argv));
}
```

`configurator` being kind of a lousy name, but I couldn't come up with something better right now.